eth/utils/address.py

```diff
@@ -1,13 +1,29 @@
 import rlp
 
 from eth_hash.auto import keccak
+from eth_typing import Address
 
+from eth.utils.hexadecimal import (
+    decode_hex,
+)
+from eth.utils.numeric import (
+    int_to_big_endian,
+)
 
-def force_bytes_to_address(value: bytes) -> bytes:
+
+def force_bytes_to_address(value: bytes) -> Address:
     trimmed_value = value[-20:]
     padded_value = trimmed_value.rjust(20, b'\x00')
-    return padded_value
+    return Address(padded_value)
+
+
+def generate_contract_address(address: Address, nonce: bytes) -> Address:
+    return Address(keccak(rlp.encode([address, nonce]))[-20:])
 
 
-def generate_contract_address(address: bytes, nonce: bytes) -> bytes:
-    return keccak(rlp.encode([address, nonce]))[-20:]
+def generate_safe_contract_address(address: Address,
+                                   salt: int,
+                                   call_data: bytes) -> Address:
+    return Address(
+        keccak(decode_hex('0xff') + address + int_to_big_endian(salt) + keccak(call_data))[-20:]
```

Comment:
This might be easier to read (and slightly more cleanly typed as) as:

```python
return force_bytes_to_address(keccak(
    decode_hex('0xff') + address + int_to_big_endian(salt) + keccak(call_data)
))
```

Also, the `decode_hex('0xff')` should probably be replaced with a constant `CREATE2_PREFIX` or just a bytes literal.

Reply:
Right! Changing it to:

```python
    return force_bytes_to_address(
        keccak(b'\xff' + address + int_to_big_endian(salt) + keccak(call_data))
    )
```

I'll also change `generate_contract_address` to use `force_bytes_to_address` to be aligned with this change.